src/RefractAST.cc

```diff
@@ -719,15 +670,30 @@ namespace drafter
                 throw std::runtime_error("Unhandled type of DataStructure");
         }
 
+        return element;
+    }
+
+    sos::Object SerializeRefract(refract::IElement* element, const refract::Registry& registry) {
+
         if (!element) {
             return sos::Object();
         }
 
+        refract::ExpandVisitor expander(registry);
```

Comment:
I don't understand, where is duplicate code?
By this solution we 
- avoid another looping trough data structures.
- minimal changes in original code as possible.